src/App.tsx

```diff
@@ -1,61 +1,119 @@
+import React, { useState } from 'react';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+/* eslint-disable */
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+/* eslint-enable */
+import { TodoList } from './components/TodoList';
+
+const getUserById = (userId: number): User | null =>
+  usersFromServer.find(user => user.id === userId) || null;
+
+const todosWithUsers: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId) || null,
+}));
 
 export const App = () => {
+  const [todos, setTodos] = useState(todosWithUsers);
+  const [userId, setUserId] = useState(0);
+  const [todoTitle, setTodoTitle] = useState('');
+  const [errorTitle, setErrorTitle] = useState('');
+  const [errorID, setErrorID] = useState('');
+
+  const addTodo = (newTodo: Todo) => {
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const createNewId = () => Math.max(...todos.map(todo => todo.id)) + 1;
+
+  const resetFields = () => {
+    setTodoTitle('');
+    setUserId(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!todoTitle) {
+      setErrorTitle('Please enter a title');
+    }
+
+    if (!userId) {
```

Comment:
Cleaner approach would be to create a helper `validate` function, not a necessity here, but still makes sense.